state/state.go

```diff
@@ -119,7 +120,24 @@ func NewWithStore(token string, cabinet store.Cabinet) (*State, error) {
 		return nil, err
 	}
 
-	return NewFromSession(s, cabinet), nil
+	return newWithAutoRescale(s, cabinet), nil
+}
+
+func newWithAutoRescale(s *session.Session, cabinet store.Cabinet) *State {
+	state := NewFromSession(s, cabinet)
+	state.ShardManager.OnShardingRequired = func() *shard.Manager {
+		state.Cabinet.Reset()
+		token := s.ShardManager.Gateways()[0].Identifier.Token
+
+		m, err := shard.NewManager(token)
+		if err != nil {
+			return nil
```

Comment:
The idea here is, that if such an error occurs, all connections are closed. If the user wants to handle that error, e.g. log it, implement a more complex retry mechanism, they would have to provide a custom implementation.
I did this, because adding another logging/capture function that will ever just receive a single error seems way to bloated, especially since the above solution is just as elegant.

Reply:
Ah right. I was reading this as part of `newWithAutoRescale` instead of in `state.ShardManager.OnShardingRequired`. Feel free to resolve this.